pkg/controller/servicebindingrequest/servicebinder.go

```diff
@@ -409,6 +409,10 @@ func buildServiceBinder(
 		options.restMapper,
 	)
 
+	if !isApplicationSelectorEmpty(options.sbr.Spec.ApplicationSelector) {
```

Comment:
This check can be delegated to `SetDefaults` because a value should always be present.

Additionally, moving this check to `SetDefaults` allows us to remove the decision branch in the main logic.

Reply:
The current code is doing what you explained. The default values are set in `SetDefaults`. There is still a possibility that the application selector could be `nil`. So, when accessing it should be checked for `nil`. Similarly `BindingPath` and `PodSpecPath` also could be `nil`.

Reply:
I'm arguing that when an initialization function such as `SetDefaults` is offered and its contents do not change during the reconciliation process, the `!isApplicationSelectorEmpty(...)` computation should always return false.

Like you've mentioned, if `BindingPath` and `PodSpecPath` could be set to `nil` by the user *and* there's a default value for both, they should be populated in `SetDefaults`.

To make this happen and since `ServiceBindingRequest.Spec.ApplicationSelector` is a value type and optional, `SetDefaults` should:

* Fill `BindingPath` field with the optimal defaults in the case the field is `nil`;
* Fill `BindingPath.PodSpecPath` with defaults in the case `PodSpecPath` is `nil`; and
* Fill `BindingPath.PodSpecPath.Containers` and `BindingPath.PodSpecPath.Volumes` in the case any of those are `nil` or `""`.

In other words, default values should be injected into the `ApplicationSelector` field in `SetDefaults` and other branches should be removed because:

* They mislead the reader, where without going further in implementation it appears the fields populated by `SetDefaults` can change during the reconciliation process, but they do not;
* More code than required for a functionality means more surface area for defects, less is always more.